kpi/management/commands/populate_kc_xform_kpi_asset_uid.py

```diff
@@ -38,48 +38,77 @@ def add_arguments(self, parser):
-        verbosity = options['verbosity']
         force = options['force']
+        chunks = options["chunks"]
+        verbosity = options['verbosity']
+        rest_service_only = options['rest_service_only']
         username = options['username']
 
+        # Counters
+        cpt = 0
+        cpt_already_populated = 0
+        cpt_failed = 0
+        cpt_patched = 0
+        cpt_no_deployments = 0
+
+        # Filter query
         query = Asset.objects
         if rest_service_only:
             query = query.exclude(hooks=None)
         if username:
             query = query.filter(owner__username=username)
 
         total = query.count()
-        cpt = 0
-        for asset in query.all():
+
+        # Use only fields we need.
+        assets = query.only('id', 'uid', '_deployment_data', 'name',
+                            'parent_id', 'owner_id')
+
+        for asset in assets.iterator(chunk_size=chunks):
             if asset.has_deployment:
                 try:
                     if asset.deployment.set_asset_uid(force=force):
                         if verbosity >= 2:
-                            print('\nAsset #{}: Patching XForm'.format(asset.id))
+                            self.stdout.write('\nAsset #{}: Patching XForm'.format(asset.id))
                         # Avoid `Asset.save()` logic. Do not touch `modified_date`
                         Asset.objects.filter(pk=asset.id).update(
                             _deployment_data=asset._deployment_data)
+                        cpt_patched += 1
                     else:
                         if verbosity >= 2:
-                            print('\nAsset #{}: Already populated'.format(asset.id))
+                            self.stdout.write('\nAsset #{}: Already populated'.format(asset.id))
+                        cpt_already_populated += 1
                 except KobocatDeploymentException as e:
                     if verbosity >= 2:
-                        print('\nERROR: Asset #{}: {}'.format(asset.id,
-                                                              str(e)))
+                        self.stdout.write('\nERROR: Asset #{}: {}'.format(asset.id,
+                                                                          str(e)))
+                    cpt_failed += 1
             else:
                 if verbosity >= 3:
-                    print('\nAsset #{}: No deployment found'.format(asset.id))
+                    self.stdout.write('\nAsset #{}: No deployments found'.format(asset.id))
+                cpt_no_deployments += 1
 
             cpt += 1
             if verbosity >= 1:
                 progress = '\rUpdated {cpt}/{total} records...'.format(
                     cpt=cpt,
                     total=total
                 )
-                sys.stdout.write(progress)
-                sys.stdout.flush()
+                self.stdout.write(progress)
+                self.stdout.flush()
 
-        print('\nDone!')
```

Comment:
Can we still say "Done!" at the end (with `self.stdout.write()`)? I wrote in the instructions that people should expect it to say "Done!" when it's finished :wink: